diff --git a/bricknil/ble_queue.py b/bricknil/ble_queue.py
index 27c2bc7..5f92ec9 100644
--- a/bricknil/ble_queue.py
+++ b/bricknil/ble_queue.py
@@ -167,7 +167,7 @@ class BLEventQ(Process):
         self.message(f"found device {self.device.name}")
 
 
-        device = bleak.BleakClient(address=self.device.address)
+        device = bleak.BleakClient(address_or_ble_device=self.device.address)
         self.devices.append(device)
         await device.connect()
 
diff --git a/bricknil/bricknil.py b/bricknil/bricknil.py
index fc042c9..f5331d7 100644
--- a/bricknil/bricknil.py
+++ b/bricknil/bricknil.py
@@ -143,12 +143,12 @@ def run(program = None): #pragma: no cover
         car.speed(0)
     bricknil.run(program())
     """
-    async def main():
+    async def main(program):
         await initialize()
         try: 
             await program()
         finally:
-            await finalize()
+            await finalize()
     loop = get_event_loop()
     loop.run_until_complete(main(program))
 
